src/components/error-overlay/error-overlay.js

```diff
@@ -0,0 +1,155 @@
+//@flow
+import style from '../../styles/style.scss';
+import {h} from 'preact';
+import {connect} from 'preact-redux';
+import {default as Icon, IconType} from '../icon';
+import BaseComponent from '../base';
+import Overlay from '../overlay';
+
+/**
+ * mapping state to props
+ * @param {*} state - redux store state
+ * @returns {Object} - mapped state to this component
+ */
+const mapStateToProps = state => ({
+  error: state.engine.error
+});
+
+const defaultError = "Something went wrong";
+const defaultSessionText = "Session ID ";
+const buttonText = "Retry";
+
+@connect(mapStateToProps)
+  /**
+   * errorOverlay component
+   *
+   * @class errorOverlay
+   * @extends {BaseComponent}
+   */
+class ErrorOverlay extends BaseComponent {
+  /**
+   * Creates an instance of ErrorObejct.
+   * @param {Object} obj obj
+   * @memberof ErrorObejct
+   */
+  constructor(obj: any) {
+    super({name: 'ErrorOverlay', player: obj.player});
+  }
+
+  /**
+   * copy input text based on input element.
+   * on success, set success internal component state for 2 seconds
+   * @returns {void}
+   * @memberof ShareOverlay
+   */
+  copyError(): void {
+    let selection = window.getSelection();
+    let dataEl = document.querySelector("." + style.errorSession);
```

Comment:
don't refer a dom element like this when u can use preact for this.
see ref attribute [https://reactjs.org/docs/refs-and-the-dom.html](url)